airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/buffers/Batch.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination_async.buffers;
+
+import io.airbyte.integrations.destination_async.GlobalMemoryManager;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import java.util.stream.Stream;
+
+/**
+ * POJO abstraction representing one discrete buffer read. This allows ergonomics dequeues by
+ * {@link io.airbyte.integrations.destination_async.FlushWorkers}.
+ * <p>
+ * The contained stream **IS EXPECTED to be a BOUNDED** stream. Returning a boundless stream has
```

Comment:
How much explaining this requires makes me think there's something we should simplify here. Not quite sure what yet though. Worth keeping in the back of our heads as we progress.

Reply:
Yeah, I thought about creating an interface called `BoundedStream` to make this clear. No google hits unfortunately.

Reply:
agreed